diff --git a/dist/index.mjs b/dist/index.mjs
index 3b94c870f00ce7e70be208199db48a3bd5569830..4aaaec7d5b656f1c4066c6295d5228d746420d17 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -24,8 +24,8 @@ const VIRTUAL_ENTRY_ALIAS = [
   /^(?:virtual:)?uno(?::(.+))?\.css(\?.*)?$/
 ];
 const LAYER_MARK_ALL = "__ALL__";
-const RESOLVED_ID_WITH_QUERY_RE = /[/\\]__uno(_.*?)?\.css(\?.*)?$/;
-const RESOLVED_ID_RE = /[/\\]__uno(?:_(.*?))?\.css$/;
+const RESOLVED_ID_WITH_QUERY_RE = /[/\\]uno(_.*?)?\.css(\?.*)?$/;
+const RESOLVED_ID_RE = /[/\\]uno(?:_(.*?))?\.css$/;
 
 const defaultPipelineExclude = [cssIdRE];
 const defaultPipelineInclude = [/\.(vue|svelte|[jt]sx|vine.ts|mdx?|astro|elm|php|phtml|html)($|\?)/];
@@ -468,7 +468,7 @@ function resolveId(id, importer) {
   for (const alias of VIRTUAL_ENTRY_ALIAS) {
     const match = id.match(alias);
     if (match) {
-      let virtual = match[1] ? `__uno_${match[1]}.css` : "__uno.css";
+      let virtual = match[1] ? `uno_${match[1]}.css` : "uno.css";
       virtual += match[2] || "";
       if (importer)
         virtual = resolve$1(importer, "..", virtual);
@@ -813,7 +813,7 @@ function GlobalModeDevPlugin(ctx) {
         const { hash, css } = await generateCSS(layer);
         return {
           // add hash to the chunk of CSS that it will send back to client to check if there is new CSS generated
-          code: `${css}__uno_hash_${hash}{--:'';}`,
+          code: `${css}uno_hash_${hash}{--:'';}`,
           map: { mappings: "" }
         };
       },
@@ -832,7 +832,7 @@ function GlobalModeDevPlugin(ctx) {
         if (layer && code.includes("import.meta.hot")) {
           let hmr = `
 try {
-  let hash = __vite__css.match(/__uno_hash_(\\w{${HASH_LENGTH}})/)
+  let hash = __vite__css.match(/uno_hash_(\\w{${HASH_LENGTH}})/)
   hash = hash && hash[1]
   if (!hash)
     console.warn('[unocss-hmr]', 'failed to get unocss hash, hmr might not work')
