diff --git a/dist/index.js b/dist/index.js
index df8cb69f40045edf5a7ae0b533f3f04b58061297..855f8a11c8de77dcb0636db8a2b519620c0e713b 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -82,7 +82,7 @@ function getScriptType() {
     return null;
   }
   const manifest = chrome.runtime.getManifest();
-  if (global.window === void 0) {
+  if (globalThis.window === void 0) {
     return "background_script" /* BACKGROUND_SCRIPT */;
   }
   if (window.location.href.startsWith(`chrome-extension://${chrome.runtime.id}`)) {